src/main/java/com/auth0/jwk/GuavaCachedJwkProvider.java

```diff
@@ -34,12 +35,24 @@ public GuavaCachedJwkProvider(final JwkProvider provider) {
      * @param size        number of jwt to cache
      * @param expiresIn   amount of time a jwk will live in the cache
      * @param expiresUnit unit of the expiresIn parameter
+     * @deprecated prefer the use of {@link GuavaCachedJwkProvider#GuavaCachedJwkProvider(JwkProvider, long, Duration)}
      */
     public GuavaCachedJwkProvider(final JwkProvider provider, long size, long expiresIn, TimeUnit expiresUnit) {
+        this(provider, size, Duration.of(expiresIn, Util.toChronoUnit(expiresUnit)));
```

Comment:
I think that instead of adding and calling this `toChronoUnit` util, we could just convert it to a unit we agree/know internally, and then use that to create the Duration object.

 i.e.
```java
long expiresInSeconds = expiresUnit.toSeconds(expiresIn);
Duration expiresInDuration = Duration.ofSeconds(expiresInSeconds);
```

What do you think? That way, the introduced utils can be removed.